src/zcash/ProvingServiceClient.cpp

```diff
@@ -12,13 +12,36 @@
 
 using namespace libzcash;
 
-ProvingServiceClient::ProvingServiceClient(std::string id, char* url, char* serverKey) : log_id(id), zmqUrl(url)
+ProvingServiceClient::ProvingServiceClient(std::string id, char* url, char* serverKey) : log_id(id), enableZAddresses(false), zmqUrl(url)
 {
     assert(strlen(serverKey) == 40);
     std::memcpy(zmqServerKey, serverKey, 41);
     assert(zmq_curve_keypair(zmqPublicKey, zmqSecretKey) == 0);
 }
 
+void ProvingServiceClient::configure_from_options()
+{
+    enableZAddresses = GetBoolArg("-icontroltheprovingserviceandwanttosenditmyzkeys", false);
+    zmqUrl = GetArg("-provingservice", "");
+
+    if (!zmqUrl.empty()) {
+        // Parameters are checked in init.cpp
+        std::memcpy(zmqServerKey, GetArg("-provingservicepubkey", "").c_str(), 41);
+        auto secretKey = GetArg("-provingserviceclientkey", "");
+        if (secretKey.empty()) {
+            assert(zmq_curve_keypair(zmqPublicKey, zmqSecretKey) == 0);
```

Comment:
does this just set `zmqPublicKey` to zero? Seems odd running it when `secretKey` is empty